src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketSearchEndpoint.java

```diff
@@ -38,8 +40,40 @@
 
     private static final Logger LOGGER = Logger.getLogger(BitbucketSearchEndpoint.class.getName());
 
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    //Fields are being injected using setters to make integration testing easier
     private BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
     private BitbucketPluginConfiguration bitbucketPluginConfiguration;
+    private HttpRequestExecutor httpRequestExecutor;
+
+    @GET
+    public HttpResponse doFindMirroredRepositories(
+            @Nullable @QueryParameter("serverId") String serverId,
+            @Nullable @QueryParameter("credentialsId") String credentialsId,
+            @Nullable @QueryParameter("repositoryId") Integer repositoryId) {
+        Jenkins.get().checkPermission(CONFIGURE);
+        if (repositoryId == null) {
+            return errorWithoutStack(
+                    HTTP_BAD_REQUEST,
+                    "Repository ID must be provided as a query parameter");
+        }
+        BitbucketPage<BitbucketMirroredRepositoryDescriptor> mirroredRepoDescriptors =
+                getMirroredRepoDescriptor(serverId, credentialsId, repositoryId);
+
+        BitbucketPage<BitbucketMirroredRepository> mirroredRepos = mirroredRepoDescriptors.transform(
+                repo -> {
+                    HttpUrl mirrorUrl = HttpUrl.parse(repo.getSelfLink());
+                    try {
+                        return httpRequestExecutor.executeGet(mirrorUrl, BitbucketCredentials.ANONYMOUS_CREDENTIALS, response -> unmarshall(response.body()));
+                    } catch (BitbucketClientException e) {
+                        LOGGER.info("Failed to retrieve repository information from mirror: " + repo.getMirrorServer().getName());
+                        return new BitbucketMirroredRepository(false, Collections.emptyMap(),
+                                repo.getMirrorServer().getName(), repositoryId, BitbucketMirroredRepositoryStatus.NOT_MIRRORED);
```

Comment:
I assume we will map this to "Upstream" and populate the list with that single option?